File: src/main/java/ie/isde/metadata/RDFNamespaces.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ie.isde.metadata;

/**
 * Enumerates the RDF namespaces required for the translated Irish Spatial Data
 * Exchange metadata
 * 
 * @author Adam Leadbetter
 * @version 1.0
 * @since 1.0
 */
public enum RDFNamespaces {
    DCT ("dct", "http://purl.org/dc/terms/"),
    /**
     * World Wide Web Consortium Data Catalog vocabulary
     */
    DCAT ("dcat", "http://www.w3.org/ns/dcat#"),
    /**
     * RDF Schema vocabulary
     */
    RDFS ("rdfs", "http://www.w3.org/1999/02/22-rdf-syntax-ns#"),
    /**
     * Schema,org vocabulary
     */
    SDO ("sdo", "https://schema.org/");
    
    private final String preferredNamespace;
    private final String uri;
    
    RDFNamespaces(String preferredNamespace, String uri){
        this.preferredNamespace = preferredNamespace;
        this.uri = uri;
    }
    /**
     * Get the preferred namespace prefix for the vocabulary
     * 
     * @return  String   The preferred namespace prefix for the vocabulary
     */
    public String ns(){return this.preferredNamespace;}
    /**
     * Get the preferred namespace URI for the vocabulary
     * 
     * @return  String   The preferred namespace URI for the vocabulary
     */
    public String nsURI(){return this.uri;}
}

File: ie/isde/RDFNamespaces.py
"""
Enumerates the Resource Description Framework and XML namespaces in use in ISDE.

Each class variable is a `dict` with two keys:
- _ns_: The preferred namespace prefix for the vocabulary as a `str`
- _url_: The URL to the vocabulary as a `str`
"""


class RDFNamespaces:
    """
    Resource Description Framework namespaces in use in the Irish Spatial Data
    Exchange

    Each class variable is a `dict` with two keys:
    - _ns_: The preferred namespace prefix for the vocabulary as a `str`
    - _url_: The URL to the vocabulary as a `str`
    """
    DCAT: dict = {'ns': 'dcat', 'url': 'http://www.w3.org/ns/dcat#'}
    """
    W3C Data Catalog vocabulary
    """
    DCT: dict = {'ns': 'dct', 'url': 'http://purl.org/dc/terms/'}
    """
    Dublin Core Terms
    """
    GSP: dict = {'ns': 'gsp', 'url': 'http://www.opengis.net/ont/geosparql#'}
    """
    Open Geospatial Consortium GeoSPARQL ontology
    """
    LOCN: dict = {'ns': 'locn', 'url': 'http://www.w3.org/ns/locn#'}
    """
    W3C Interoperable Solutions for Administrations (ISA) Programme Location Core Vocabulary
    """
    RDF: dict = {'ns': 'rdf', 'url': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'}
    """
    W3C Resource Description Framework concepts vocabulary
    """
    RDFS: dict = {'ns': 'rdfs', 'url': 'http://www.w3.org/2000/01/rdf-schema#'}
    """
    W3C Resource Description Framework Schema vocabulary
    """
    SDO: dict = {'ns': 'sdo', 'url': 'https://schema.org/'}
    """
    Schema.org vocab
    """